core_types/src/entry/cap_entries.rs

```diff
@@ -0,0 +1,250 @@
+use crate::{
+    cas::content::{Address, AddressableContent},
+    dna::capabilities::{CallSignature, CapabilityType},
+    entry::Entry,
+    error::HolochainError,
+    json::JsonString,
+};
+
+pub type CapTokenValue = Address;
+
+/// System entry to hold a capability token for use as a caller
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, DefaultJson)]
+pub struct CapToken {
+    token: CapTokenValue,
+}
+
+impl CapToken {
+    pub fn new(token: CapTokenValue) -> Self {
+        CapToken { token }
+    }
+    pub fn token(self) -> CapTokenValue {
+        self.token
+    }
+}
+
+/// System entry to hold a capabilities granted by the callee
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, DefaultJson)]
+pub struct CapTokenGrant {
+    assignees: Option<Vec<Address>>,
+}
+
+impl CapTokenGrant {
+    pub fn new(assignees: Option<Vec<Address>>) -> Self {
```

Comment:
Does this want to be private then, since it looks like the proper constructor is `create`? Maybe even rename `create` to `new`? (Only the tests are using `new` now.)